server/data/userClient.js

```diff
@@ -26,6 +27,40 @@ module.exports = {
     return rows.map(convertPropertyNames)
   },
 
+  async getIncompleteRoUsers() {
+    const query = {
+      text: `select
+               timestamp as sent_timestamp,
+               details -> 'bookingId' as booking_id,
+               details -> 'submissionTarget' -> 'com' -> 'deliusId' as sent_staffcode,
+               details -> 'submissionTarget' -> 'com' -> 'name' as sent_name,
+               (s.staff_id is not null) as mapped,
+               s.auth_onboarded,
+               s.first_name, s.last_name, s.nomis_id
+             from (
+                    select *,
+                           row_number() over (
+                             partition by (a.details -> 'submissionTarget' -> 'com' -> 'deliusId')
+                             order by a.timestamp ASC
+                             ) as rownbr
+                    from audit a
+                           left join staff_ids s
+                                     on s.staff_id = (a.details-> 'submissionTarget' -> 'com' ->> 'deliusId')
+                    where a.action = 'SEND'
```

Comment:
What are the -> and ->> operators here? Not come across these before... I could google..

Reply:
it's postgres JSON. Querying inside a json object. Quite ugly